### ISSUE TITLE
iTunes Connect - Binary Rejected: 2. 1 PERFORMANCE: APP COMPLETENESS

### ISSUE THREAD

**Issue by StasDoskalenko** [facebook/react-native#12168]
### Description

Hi everyone! Tried to upload the application to the iTunes connect for External Beta Review. However, I got the **binary rejection** with following reason:

> We were unable to review your app as it crashed on launch. We have attached detailed crash logs to help troubleshoot this issue.

The main problem I discovered in the crashlog is:

> Application Specific Information:
abort() called

Crashlogs:
[attachment-7163109193754095243crashlog-DB2EC298-3999-4762-A799-846CBA673FA7.txt](https://github.com/facebook/react-native/files/748046/attachment-7163109193754095243crashlog-DB2EC298-3999-4762-A799-846CBA673FA7.txt)
[attachment-9210219322814044346crashlog-BDD1F2F3-E3D3-4E5E-8A96-2E842CA9F3D9.txt](https://github.com/facebook/react-native/files/748047/attachment-9210219322814044346crashlog-BDD1F2F3-E3D3-4E5E-8A96-2E842CA9F3D9.txt)
[attachment-6841430649826099204crashlog-B3B5CD65-0C83-446C-98E7-D0A5A316DAB0.txt](https://github.com/facebook/react-native/files/748048/attachment-6841430649826099204crashlog-B3B5CD65-0C83-446C-98E7-D0A5A316DAB0.txt)

### Possible solution

I tried to add **react-native-splash-screen** module to avoid this crash. But still got rejection but without crash log:

> We discovered one or more bugs in your app when reviewed on iPad and iPhone running iOS 10.2.1 on Wi-Fi connected to an IPv6 network.
Specifically, we found the app did not progress further than the splash screen.
Please see attached screenshots for details.

Tried to test within ip v6 network. All seems to be fine.

Need help! Thanks.

### Additional Information

* React Native version: 0.40
* Platform: iOS
* Operating System: MacOS

**Comment by 7ynk3r** [comment]
hi @StasDoskalenko! im having a similar problem. what was the solution in your case?

**Comment by StasDoskalenko** [comment]
Hi @7ynk3r ! Upgraded to the last version of RN, added **splash screen**, built with bitcode, and error has gone.

**Comment by 7ynk3r** [comment]
Thank you so much @StasDoskalenko!

**Comment by spurton** [comment]
@StasDoskalenko What did you use to add a splash screen? I am having a similar error...

**Comment by Ethan0007** [comment]
@StasDoskalenko , sir what version are you using? we are using 0.43 also we are using Firebase but our app still rejected in iTunes.

**Comment by brownieboy** [comment]
@Ethan0007,

Did you ever find a solution?

I'm also using Firebase and am also getting rejected by iTunes connect.  The app hangs on a blank screen for them, but not for me.  I've tested on an IPV6 network using Apple's instructions.

**Comment by Ethan0007** [comment]
Yes sir, please try to resolved warnings and errors in your application.

**Comment by brownieboy** [comment]
@Ethan0007,

Well, that's always good advice!!  ;)

I have no errors in my application.  There are warnings from 3rd party libraries that I can't fix myself.

Do you happen to remember exactly what it was that fixed it for you?

**Comment by Ethan0007** [comment]
review all your codes or function that will generate warnings or error , like for example unhandled exception for ajax request.

**Comment by brownieboy** [comment]
I submitted a new build to Apple last night, and it was approved for beta testing this morning.

There were a few changes:
- I upgraded to later versions of some libraries, including React Native itself and React Navigation
- I fixed some warnings in my own code my removing some duplicate font files that were being loaded multiple times in the Xcode Build settings.
- Fixed some warnings about possibly unhandled Promise rejections

Really, would any that caused my app to hang on a white screen?  And if so, why didn't it hang in my own environment, even after I'd sent the app through TestFlight, tested on an IPV6 network and so on.

It all seems so arbitrary.  Still, a huge relief to get my first pass from Apple!!
